storage/innobase/srv/srv0srv.cc

```
@@ -2761,6 +2770,199 @@ srv_enable_temp_encryption_if_set()
 	}
 }
 
+/** Enable the undo log encryption if it is set.
+It will try to enable the undo log encryption and write the metadata to
+undo log file header, if innodb_undo_log_encrypt is ON. */
+static
+void
+srv_enable_undo_encryption_if_set()
+{
+	fil_space_t*	space;
+	const char*	cant_set_undo_tablespace = "Can't set undo tablespace";
+	const char*	to_be_encrypted = " to be encrypted";
+
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	/* Check if encryption for undo log is enabled or not. If it's
+	enabled, we will store the encryption metadata to the space header
+	and start to encrypt the undo log block from now on. */
+	if (srv_undo_log_encrypt) {
+		if (undo::spaces->empty()) {
+			srv_undo_log_encrypt = false;
+			ib::error() << cant_set_undo_tablespace << "s"
+				<< to_be_encrypted
+				<< ", since innodb_undo_tablespaces=0.";
```

Can you please show where this is tested? In MTR I see that this error is being suppressed in one of the testcases, but I was unable to find an attempt to execute this path.